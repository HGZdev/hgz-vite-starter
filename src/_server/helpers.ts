import jwt from "jsonwebtoken";

const JWT_SECRET = process.env.JWT_SECRET || "secret";
console.log("JWT_SECRET:", JWT_SECRET);

export const getUserFromToken = (token: string) => {
  let user;
  if (token) {
    try {
      user = jwt.verify(token, JWT_SECRET);
      return user;
    } catch (err) {
      console.log(err);
    }
  }
  return user;
};

export const makeTokenFromUser = (user: unknown, expiresIn: string = "1hr") => {
  const token = jwt.sign({user}, JWT_SECRET, {
    expiresIn,
  });

  return token;
};
