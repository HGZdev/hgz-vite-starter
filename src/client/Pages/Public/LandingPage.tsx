import React from "react";
import {useNavigate} from "react-router-dom";
import {ButtonPrimary, ButtonSecondary} from "../../Components/Buttons";
import {Avatar} from "../../Components/Avatar";

const LandingPage: React.FC = () => {
  const navigate = useNavigate();

  return (
    <div
      data-testid="LandingPage"
      className=" flex flex-col items-center justify-center h-screen gap-4"
    >
      <Avatar />
      <h1 className="text-4xl font-sans text-center">
        Welcome to My Awesome Starter! ☕️
      </h1>
      <p className="text-2xl font-sans">Explore, Learn, and Enjoy Your Stay!</p>
      <div className="flex justify-center gap-4">
        <ButtonSecondary
          className="btn-block"
          onClick={() => navigate("/registration")}
          aria-label="Register"
        >
          Register
        </ButtonSecondary>
        <ButtonPrimary
          className="btn-block"
          onClick={() => navigate("/dashboard")}
          aria-label="Login"
        >
          Login
        </ButtonPrimary>
      </div>
    </div>
  );
};

export default LandingPage;
