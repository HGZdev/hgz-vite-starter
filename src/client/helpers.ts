import {serialize} from "cookie";

export const setCookie = (
  name: string,
  value: string,
  expiryHours: number
): string => {
  const expires = new Date();
  expires.setTime(expires.getTime() + expiryHours * 60 * 60 * 1000);

  const cookie = serialize(name, value, {
    expires,
    path: "/",
  });

  document.cookie = cookie;
  return cookie;
};

export const destroyCookie = (name: string): string => {
  const expires = new Date();
  expires.setTime(expires.getTime() - 1);

  const cookie = serialize(name, "", {
    expires,
    path: "/",
  });

  document.cookie = cookie;
  return cookie;
};
