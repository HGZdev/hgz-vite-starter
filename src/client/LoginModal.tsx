import React, {useState} from "react";
import {useLogin} from "../_server/queries";
import styled from "styled-components";

const ModalBackground = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgba(0, 0, 0, 0.8);
`;

const ModalContent = styled.div`
  background-color: white;
  padding: 2rem;
  border-radius: 0.5rem;
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.2);
  width: 100%;
  max-width: 24rem;
`;

const FormGroup = styled.div`
  margin-bottom: 1.5rem;
`;

const Label = styled.label`
  display: block;
  margin-bottom: 0.5rem;
  font-weight: bold;
  color: #4a5568;
`;

const Input = styled.input`
  width: calc(100% - 1rem);
  padding: 0.5rem;
  border: 1px solid #e2e8f0;
  border-radius: 0.25rem;
`;

const ErrorMessage = styled.div`
  color: #ef4444;
  margin-bottom: 1rem;
`;

const SubmitButton = styled.button`
  width: 100%;
  padding: 1rem;
  background-color: #2563eb;
  color: white;
  border-radius: 0.25rem;
  transition: background-color 0.3s ease;

  &:hover {
    background-color: #1d4ed8;
  }
`;

interface LoginModalProps {
  onClose: () => void;
}

const LoginModal: React.FC<LoginModalProps> = ({onClose}) => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [errorMessage, setErrorMessage] = useState<string | null>(null);

  const [login] = useLogin();

  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    setErrorMessage(null);
    try {
      const {data} = await login({email, password});
      const token = data?.login?.token;

      if (token) {
        // setCookie("token", token, 1);
        onClose();
      }
    } catch (err) {
      if (err instanceof Error) setErrorMessage(err.message); // Error handling
    }
  };

  return (
    <ModalBackground>
      <ModalContent>
        <form onSubmit={handleSubmit}>
          <FormGroup>
            <Label htmlFor="email">Email:</Label>
            <Input
              type="email"
              id="email"
              value={email}
              onChange={(e) => setEmail(e.target.value)}
              required
            />
          </FormGroup>
          <FormGroup>
            <Label htmlFor="password">Password:</Label>
            <Input
              type="password"
              id="password"
              value={password}
              onChange={(e) => setPassword(e.target.value)}
              required
            />
          </FormGroup>
          {errorMessage && <ErrorMessage>{errorMessage}</ErrorMessage>}
          <SubmitButton type="submit">Login</SubmitButton>
        </form>
      </ModalContent>
    </ModalBackground>
  );
};

export default LoginModal;
