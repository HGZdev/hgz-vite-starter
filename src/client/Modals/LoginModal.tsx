import React, {useState} from "react";
import * as Yup from "yup";
import {InputField} from "../Form/Form";
import {Form, Formik} from "formik";
import {useLogin} from "../../_server/queries";
import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
  Typography,
  Box, // Import Box from @mui/material to use flexbox
} from "@mui/material";
import {Link} from "react-router-dom";

// Define validation schema
const validationSchema = Yup.object().shape({
  email: Yup.string().required("Email is required"),
  password: Yup.string().required("Password is required"),
});

// Define types for user data
interface UserData {
  email: string;
  password: string;
}

// Define the LoginForm component
interface LoginFormProps {
  onClose: () => void;
}

const LoginForm: React.FC<LoginFormProps> = ({onClose}) => {
  const [login] = useLogin();
  const [loginError, setLoginError] = useState<string | null>(null);

  return (
    <Formik<UserData>
      initialValues={{
        email: "",
        password: "",
      }}
      validationSchema={validationSchema}
      onSubmit={async (values) => {
        try {
          const {data} = await login(values);
          const token = data?.login?.token;

          if (token) {
            onClose();
          }
        } catch (error) {
          setLoginError("Something went wrong");
          console.error("Error during login:", error);
        }
      }}
    >
      <Form>
        <InputField
          label="Email"
          name="email"
          type="email"
          autoComplete="email"
        />
        <InputField
          label="Password"
          name="password"
          type="password"
          autoComplete="current-password"
        />
        {loginError && (
          <Typography variant="caption" color="error">
            {loginError}
          </Typography>
        )}
        <DialogActions>
          <Box flexGrow={1}>
            <Link to="/registration">
              <Button color="primary">Sign up</Button>
            </Link>
          </Box>
          <Button onClick={onClose} color="secondary">
            Cancel
          </Button>
          <Button type="submit" variant="contained" color="primary">
            Login
          </Button>
        </DialogActions>
      </Form>
    </Formik>
  );
};

interface LoginModalProps {
  onClose: () => void;
  open: boolean;
}

const LoginModal: React.FC<LoginModalProps> = ({onClose, open}) => {
  return (
    <Dialog open={open} onClose={onClose} maxWidth="sm" fullWidth>
      <DialogTitle>Login</DialogTitle>
      <DialogContent>
        <LoginForm onClose={onClose} />
      </DialogContent>
    </Dialog>
  );
};

export default LoginModal;
