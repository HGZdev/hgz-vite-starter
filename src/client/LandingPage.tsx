import React, {useState} from "react";
import styled from "styled-components";
import {useGetUserMe} from "../_server/queries";
import LoginModal from "./LoginModal";
import Dashboard from "./Dashboard";

const LandingPageContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100vh;
`;

const SignupButton = styled.button`
  padding: 10px 20px;
  font-size: 16px;
  background-color: #007bff;
  color: #fff;
  border: none;
  border-radius: 4px;
  cursor: pointer;
`;

const LoginButton = styled.button`
  margin-top: 10px;
  padding: 10px 20px;
  font-size: 16px;
  background-color: #28a745;
  color: #fff;
  border: none;
  border-radius: 4px;
  cursor: pointer;
`;

const LandingPage: React.FC = () => {
  const [showLoginModal, setShowLoginModal] = useState(false);
  const {data, loading, error} = useGetUserMe();

  if (!data && loading) return <p>Loading...</p>;
  if (error) return <p>Error: {error.message}</p>;

  const getUserMe = data?.getUserMe;

  if (showLoginModal)
    return <LoginModal onClose={() => setShowLoginModal(false)} />;

  return (
    <LandingPageContainer>
      {!getUserMe ? (
        <>
          <SignupButton
            aria-label="Sign up"
            onClick={() => (window.location.href = "/registration")}
          >
            Sign up
          </SignupButton>
          <LoginButton
            aria-label="Login"
            onClick={() => setShowLoginModal(true)}
          >
            Log in
          </LoginButton>
        </>
      ) : (
        <Dashboard />
      )}
    </LandingPageContainer>
  );
};

export default LandingPage;
