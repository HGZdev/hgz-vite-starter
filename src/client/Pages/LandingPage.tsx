import React, {useState} from "react";
import {Button, CircularProgress, Typography, Avatar} from "@mui/material";
import {useGetUserMe} from "../../_server/queries";
import LoginModal from "../Modals/LoginModal";
import Dashboard from "./Dashboard";
import styled from "styled-components";
import {useNavigate} from "react-router-dom";

const LandingPageContainer = styled("div")`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100vh;
`;

const ButtonContainer = styled("div")`
  display: flex;
  flex-direction: row; /* Change this line to set buttons in one line */
  align-items: center;
  gap: 10px;
  margin-top: 20px;
`;

const SignupButton = styled(Button)``;
const LoginButton = styled(Button)``;

const LandingPage: React.FC = () => {
  const [showLoginModal, setShowLoginModal] = useState(false);
  const {data, loading, error} = useGetUserMe();
  const navigate = useNavigate();

  if (!data && loading)
    return <CircularProgress data-testid="loading" aria-label="loading" />;
  if (error) return <Typography>Error: {error.message}</Typography>;

  const getUserMe = data?.getUserMe;
  if (showLoginModal) {
    return (
      <LoginModal
        open={showLoginModal}
        onClose={() => setShowLoginModal(false)}
      />
    );
  }
  return (
    <LandingPageContainer data-testid="LandingPage">
      {!getUserMe ? (
        <>
          <Avatar
            alt="Favicon"
            src="/../assets/favicon.ico"
            sx={{width: 250, height: 250}}
          />
          <Typography>Welcome!</Typography>
          <ButtonContainer>
            <SignupButton
              variant="contained"
              color="primary"
              onClick={() => navigate("/registration")}
              aria-label="Register"
            >
              Register
            </SignupButton>
            <LoginButton
              variant="contained"
              color="secondary"
              onClick={() => setShowLoginModal(true)}
              aria-label="Login"
            >
              Login
            </LoginButton>
          </ButtonContainer>
        </>
      ) : (
        <Dashboard />
      )}
    </LandingPageContainer>
  );
};

export default LandingPage;
