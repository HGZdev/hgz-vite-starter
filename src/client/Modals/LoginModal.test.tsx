// LoginModal.test.tsx
import {render, screen, waitFor} from "@testing-library/react";
import userEvent, {type UserEvent} from "@testing-library/user-event";
import {describe, test, beforeEach, expect, vi} from "vitest";
import {mockServer} from "../../tests/vitestSetup";
import {
  counterIncrementingRes,
  getUserMeNotLoggedInRes,
} from "../../tests/graphqlHandlers";
import {MockedRoot} from "../../tests/testing-library/Components";
import {findBtn, findId, findText} from "../../tests/testing-library/helpers";
import {Route} from "react-router-dom";
import LoginModal from "./LoginModal";
import LandingPage from "../Pages/LandingPage";

describe("LoginModal Component Tests", () => {
  let onCloseMock: () => void;
  let user: UserEvent;
  beforeEach(() => {
    onCloseMock = vi.fn();
    user = userEvent.setup();
    mockServer.use(...getUserMeNotLoggedInRes, ...counterIncrementingRes);
  });

  test("go to LoginModal, than go back when Cancel button is clicked", async () => {
    render(
      <MockedRoot>
        <Route path="/" element={<LandingPage />} />
      </MockedRoot>
    );

    expect(screen.getByTestId("loading"));
    expect(await findBtn("Login"));

    await user.click(await findBtn("Login"));

    await waitFor(async () => {
      expect(await findId("LoginModal")).toMatchSnapshot();
      expect(await findBtn("Register"));
      expect(await findBtn("Cancel"));
      expect(await findBtn("Login"));

      await user.click(screen.getByRole("button", {name: "Cancel"}));
      expect(await findId("LandingPage"));
    });
  });

  test("calls onClose when Cancel button is clicked", async () => {
    render(
      <MockedRoot>
        <Route path="/" element={<LoginModal open onClose={onCloseMock} />} />
      </MockedRoot>
    );

    expect(await findBtn("Cancel"));
    await user.click(await findBtn("Cancel"));
    expect(onCloseMock).toHaveBeenCalledTimes(1);
  });

  test("displays validation errors for empty fields on form submission", async () => {
    render(
      <MockedRoot>
        <Route path="/" element={<LoginModal open onClose={onCloseMock} />} />
      </MockedRoot>
    );
    user.click(await findBtn("Login"));

    await waitFor(async () => {
      expect(await findText("Email is required"));
      expect(await findText("Password is required"));
    });
  });

  // test("handles login error and displays error message", async () => {
  //   // Mocking login function to simulate an error
  //   vi.mock("../../_server/queries", () => ({
  //     ...vi.requireActual("../../_server/queries"),
  //     useLogin: () => [
  //       async () => {
  //         throw new Error("Mock login error");
  //       },
  //     ],
  //   }));

  //   render(<LoginModal open onClose={onCloseMock} />);

  //   await act(async () => {
  //     userEvent.type(screen.getByLabelText("Email"), "test@example.com");
  //     userEvent.type(screen.getByLabelText("Password"), "password123");
  //     userEvent.click(screen.getByRole("button", {name: "Login"}));
  //   });

  //   await waitFor(() => {
  //     expect(onCloseMock).not.toHaveBeenCalled();
  //     expect(screen.getByText("Something went wrong"));
  //   });
  // });

  // test("submits login form successfully and calls onClose", async () => {
  //   // Mocking login function to simulate a successful login
  //   vi.mock("../../_server/queries", () => ({
  //     ...vi.requireActual("../../_server/queries"),
  //     useLogin: () => [async () => ({data: {login: {token: "mockToken"}}})],
  //   }));

  //   render(<LoginModal open onClose={onCloseMock} />);

  //   await act(async () => {
  //     userEvent.type(screen.getByLabelText("Email"), "test@example.com");
  //     userEvent.type(screen.getByLabelText("Password"), "password123");
  //     userEvent.click(screen.getByRole("button", {name: "Login"}));
  //   });

  //   await waitFor(() => {
  //     expect(onCloseMock).toHaveBeenCalled();
  //   });
  // });
});
