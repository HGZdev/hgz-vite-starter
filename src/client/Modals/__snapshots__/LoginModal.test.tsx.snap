// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoginModal Component Tests > calls onClose when Cancel button is clicked > true 1`] = `
<div
  class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
  data-testid="LoginModal"
  role="presentation"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-testid="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
    role="presentation"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      aria-labelledby=":r3:"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullWidth css-tlc64q-MuiPaper-root-MuiDialog-paper"
      role="dialog"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
        id=":r3:"
      >
        Login
      </h2>
      <div
        class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
      >
        <form
          action="#"
        >
          <div
            class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for=":r4:"
              id=":r4:-label"
            >
              Email
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                autocomplete="email"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r4:"
                name="email"
                type="email"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-14lo706"
                >
                  <span>
                    Email
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiTypography-root MuiTypography-caption css-19m5e5t-MuiTypography-root"
          />
          <div
            class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for=":r5:"
              id=":r5:-label"
            >
              Password
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                autocomplete="current-password"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r5:"
                name="password"
                type="password"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-14lo706"
                >
                  <span>
                    Password
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiTypography-root MuiTypography-caption css-19m5e5t-MuiTypography-root"
          />
          <div
            class="MuiDialogActions-root MuiDialogActions-spacing css-knqc4i-MuiDialogActions-root"
          >
            <div
              class="MuiBox-root css-i9gxme"
            >
              <a
                href="/registration"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  Register
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </a>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium css-1d3f8j8-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Cancel
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              Login
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
  <div
    data-testid="sentinelEnd"
    tabindex="0"
  />
</div>
`;

exports[`LoginModal Component Tests > go to LoginModal, than go back when Cancel button is clicked 1`] = `
<div
  class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
  data-testid="LoginModal"
  role="presentation"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-testid="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
    role="presentation"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      aria-labelledby=":r0:"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullWidth css-tlc64q-MuiPaper-root-MuiDialog-paper"
      role="dialog"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
        id=":r0:"
      >
        Login
      </h2>
      <div
        class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
      >
        <form
          action="#"
        >
          <div
            class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for=":r1:"
              id=":r1:-label"
            >
              Email
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                autocomplete="email"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r1:"
                name="email"
                type="email"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-14lo706"
                >
                  <span>
                    Email
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiTypography-root MuiTypography-caption css-19m5e5t-MuiTypography-root"
          />
          <div
            class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for=":r2:"
              id=":r2:-label"
            >
              Password
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                autocomplete="current-password"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r2:"
                name="password"
                type="password"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-14lo706"
                >
                  <span>
                    Password
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiTypography-root MuiTypography-caption css-19m5e5t-MuiTypography-root"
          />
          <div
            class="MuiDialogActions-root MuiDialogActions-spacing css-knqc4i-MuiDialogActions-root"
          >
            <div
              class="MuiBox-root css-i9gxme"
            >
              <a
                href="/registration"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  Register
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </a>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium css-1d3f8j8-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Cancel
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              Login
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
  <div
    data-testid="sentinelEnd"
    tabindex="0"
  />
</div>
`;
